### PR TITLE
soft-serve: init at 0.1.0

### PR DESCRIPTION
**Pull request by penguwin**
###### Motivation for this change

Add soft-serve, a tasty, self-hosted Git server for the command line to the nixpkgs collection 🍦

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/soft-serve/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "soft-serve";
+    rev = "v${version}";
+    sha256 = "0z88699q34a9cbhcz12y2qs2qrspfd8yx4ay0r8jzvkgax9ylrlk";
+  };
+
+  vendorSha256 = "1g2iznfh08l23i81x7g2bhc7l8cppshzlyynxik4jshswlpv80sr";
+
+  doCheck = false;
+
+  ldflags = [ "-s" "-w" "-X=main.Version=${version}" ];
```
Comment (legendofmiracles) [inline pkgs/servers/soft-serve/default.nix:19]:
```suggestion
  ldflags = [ "-s" "-w" "-X=main.Version=${version}" ];
  
  nativeBuildInputs = [ makeWrapper ];
  
  postInstall = ''
  	wrapProgram $out/bin/soft \
  		--prefix PATH : "${lib.makeBinPath [ git ]}"
  '';
```

File: pkgs/servers/soft-serve/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (legendofmiracles) [inline pkgs/servers/soft-serve/default.nix:2]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, makeWrapper, git }:
```

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

**Comment by penguwin** [comment]
@legendofmiracles Thanks for the suggestions 💪 !

**Comment by penguwin** [comment]
Heh, seems like githubs code suggestions doesn't comply to the editorconfig settings 😅 quickly fixed that

**Comment by legendofmiracles** [comment]
> @legendofmiracles Thanks for the suggestions 💪 !


No problem, really awesome tool! Can you squash the commit made by my changes into your other commit?

**Comment by penguwin** [comment]
Nice to hear 😊 Yup, squashed the commits 🤙 thanks for reviewing this!

**Review by legendofmiracles — APPROVED** [review]

**Pull request merged**